diff --git a/src/util/Btrfs.cpp b/src/util/Btrfs.cpp
index 1c594fa..2f5d73d 100644
--- a/src/util/Btrfs.cpp
+++ b/src/util/Btrfs.cpp
@@ -71,7 +71,7 @@ QStringList Btrfs::children(const uint64_t subvolId, const QString &uuid) const
 
     QStringList children;
 
-    while (returnCode != BTRFS_UTIL_ERROR_STOP_ITERATION) {
+    while (returnCode == BTRFS_UTIL_OK) {
         char *path = nullptr;
         struct btrfs_util_subvolume_info subvolInfo;
         returnCode = btrfs_util_subvolume_iterator_next_info(iter, &path, &subvolInfo);
@@ -265,7 +265,7 @@ void Btrfs::loadSubvols(const QString &uuid)
 
         SubvolumeMap subvols;
 
-        while (returnCode != BTRFS_UTIL_ERROR_STOP_ITERATION) {
+        while (returnCode == BTRFS_UTIL_OK) {
             char *path = nullptr;
             struct btrfs_util_subvolume_info subvolInfo;
             returnCode = btrfs_util_subvolume_iterator_next_info(iter, &path, &subvolInfo);
